corehq/apps/app_manager/app_translations/parser.py

```diff
@@ -0,0 +1,112 @@
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _type, _sheet_name, _unique_id = re.match(context_regex, context).groups()
+            ws.append([_type, _sheet_name, po_entry.msgid, po_entry.msgstr, _unique_id])
+            self.translations[ws.title].append(
+                {
+                    'Type': _type,
+                    'sheet_name': _sheet_name,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr,
+                    'unique_id': _unique_id
+                }
+            )
+
+    def _parse_module_sheet(self, ws, po_entries):
+        # expected context format
+        # index: case property: list/detail
+        context_regex = r'^(\d+):(.+):(list|detail)$'
+        ws.append(["case_property", "list_or_detail", self.key_lang_str, self.source_lang_str])
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _case_property, _list_or_detail = re.match(context_regex, context).groups()
+            ws.append([_case_property, _list_or_detail, po_entry.msgid, po_entry.msgstr])
+            self.translations[ws.title].append(
+                {
+                    'case_property': _case_property,
+                    'list_or_detail': _list_or_detail,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr
+                }
+            )
+
+    def _parse_form_sheet(self, ws, po_entries):
+        # expected context regex
+        # index: label
+        context_regex = r'^(\d+):(.+)$'
+        ws.append(["label", self.key_lang_str, self.source_lang_str])
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _label = re.match(context_regex, context).groups()
+            ws.append([_label, po_entry.msgid, po_entry.msgstr])
+            self.translations[ws.title].append(
+                {
+                    'label': _label,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr
+                }
+            )
+
+    def result_file_name(self, version):
+        return ("TransifexTranslations {}-{}:v{} {}.xlsx".format(
+            self.transifex.key_lang, self.transifex.source_lang,
+            version, datetime.utcnow().strftime(DATETIME_FORMAT))
+        )
+
+    def _generate_sheets(self, wb, resource_slugs):
+        version = self.transifex.version
+        all_translations = self.transifex.get_translations(version, resource_slugs)
+        for resource_name in all_translations:
+            po_entries = all_translations[resource_name]
+            sheet_name = resource_name.split("_v%s" % version)[0]
+            self.translations[sheet_name] = []
+            ws = wb.create_sheet(title=sheet_name)
+            self._parse_sheet(ws, po_entries)
+
+    def generate_excel_file(self, resource_slugs=None):
```

Comment:
Ya, its more of a proactive addition because it would probably come in handy to pull specific resource slugs.  I used it during testing so i left it there for now. Happy to remove if you think its not necessary/YAGNI.